CRM/Dedupe/Finder.php

```diff
@@ -171,11 +171,9 @@ private static function appendCustomDataFields(array &$fields, array &$flat, str
     $subTypes = $fields['contact_sub_type'] ?? [];
     // Only return custom for subType + unrestricted or return all custom
     // fields.
-    $tree = self::getTree($ctype, $subTypes, $fields);
-    foreach ($tree as $key => $cg) {
-      foreach ($cg['fields'] as $cf) {
-        $flat[$cf['column_name']] = $cf['customValue']['data'] ?? NULL;
-      }
+    $customFields = self::getTree($ctype, $subTypes, $fields);
```

Comment:
Would be nice if this could just be switched from `getTree()` to `BAO_CustomGroup::getAll()` - is that the eventual goal?